aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -534,7 +534,8 @@ TORCH_IMPL_FUNC(addmv_out_cuda)(const Tensor &self, const Tensor &mat, const Ten
 
       // Check for contiguity of `vec` and update `vec_stride` accordingly
       const auto vec_contiguous = vec_stride == 0 ? vec.contiguous() : vec;
-      vec_stride = vec_contiguous.stride(0);
+      // A vector can be contiguous and have a stride of zero if it has it is of length 1
+      vec_stride = std::max<int64_t>(vec_contiguous.stride(0), 1LL);
```

Comment:
That's an interesting situation. Here's what I tried to reproduce the problem described in the comment:
```py
In [1]: import torch
In [2]: a = torch.randn(()) # a has shape ()
In [3]: b = a.expand(1) # b has shape (1,)
In [4]: b.stride()
Out[4]: (0,)
In [5]: b.is_contiguous()
Out[5]: True
In [6]: b.contiguous().stride()
Out[6]: (0,)
```

Reply:
Fwiw, this was fixed in one of the PRs of https://github.com/pytorch/pytorch/pull/75279, only that natalia wanted to land this quicker, so she carved out some of yhe PRs from the stack.

On a different note, this is indeed a funny edge case, where a tensor with a stride of 0 can be contiguous. I'm quite sure that pretty much all the linalg functions would fail as currently implemented on inputs like these.
Note that, more generally, a problematic input can be any input where one of its dimensions is 1 and the stride of that dimension is zero.